baseplate/context/cassandra.py

```diff
@@ -30,28 +32,57 @@ def cluster_from_config(app_config, prefix="cassandra.", **kwargs):
     * ``contact_points`` (required): comma delimited list of contact points to
       try connecting for cluster discovery
     * ``port``: The server-side port to open connections to.
+    * ``credentials_secret`` (optional): the key used to retrieve the database
+        credentials from ``secrets`` as a :py:class:`~baseplate.secrets.CredentialSecret`.
 
     """
     assert prefix.endswith(".")
-    config_prefix = prefix[:-1]
-    cfg = config.parse_config(
-        app_config,
+    parser = config.SpecParser(
         {
-            config_prefix: {
-                "contact_points": config.TupleOf(config.String),
-                "port": config.Optional(config.Integer, default=None),
-            }
-        },
+            "contact_points": config.TupleOf(config.String),
+            "port": config.Optional(config.Integer, default=None),
+            "credentials_secret": config.Optional(config.String),
+        }
     )
-
-    options = getattr(cfg, config_prefix)
+    options = parser.parse(prefix[:-1], app_config)
 
     if options.port:
         kwargs.setdefault("port", options.port)
 
+    if options.credentials_secret:
+        if not secrets:
+            raise TypeError("'secrets' is required if 'credentials_secret' is set")
+        credentials = secrets.get_credentials(options.credentials_secret)
+        kwargs.setdefault(
+            "auth_provider",
+            PlainTextAuthProvider(username=credentials.username, password=credentials.password),
+        )
+
     return Cluster(options.contact_points, **kwargs)
 
 
+class CassandraClient(config.Parser):
+    """Configure a Cassandra client.
+
+    This is meant to be used with
+    :py:meth:`baseplate.core.Baseplate.configure_context`.
+
+    See :py:func:`cluster_from_config` for available configurables.
+
+    :param keyspace: Which keyspace to set as the default for operations.
+
+    """
+
+    def __init__(self, keyspace: str, **kwargs):
+        self.keyspace = keyspace
+        self.kwargs = kwargs
+
+    def parse(self, key_path: str, raw_config: config.RawConfig) -> ContextFactory:
+        cluster = cluster_from_config(raw_config, prefix=f"{key_path}.", **self.kwargs)
+        session = cluster.connect(keyspace=self.keyspace)
```

Comment:
I looked into it and confused myself thinking that was covered by the kwargs thing but now I see that's an attribute on session instead. I'll get that in, my bad. That said, it looks like we're supposed to use this now? https://docs.datastax.com/en/developer/python-driver/3.18/execution_profiles/

Reply:
Interesting, we'd have to be able to force a minimum version but seems reasonable?

Reply:
Looks like it lost "experimental" status in 3.13 (Jan 2018) https://github.com/datastax/python-driver/blob/master/CHANGELOG.rst#3130

Our wheelhouse has 3.4, 3.13, 3.16, 3.17 so it's really only the 3.4s that will have issues. Sounds reasonable to build on this instead of the older method.

Reply:
OK, I made it possible to pass execution profiles in.